src/org/zaproxy/zap/extension/cspscanner/ContentSecurityPolicyScanner.java

```diff
@@ -0,0 +1,305 @@
+		return returnSb.toString();
+	}
+
+	/**
+	 * Extracts a list of headers, and returns them without changing
+	 * their cases.
+	 * @param msg HTTP Response message
+	 * @param header The header field(s) to be found
+	 * @return list of the matched headers
+	 */
+	private List<String> getHeaderField(HttpMessage msg, String header) {
+		List<String> matchedHeaders = new ArrayList<>();
+		String headers = msg.getResponseHeader().toString();
+		String[] headerElements = headers.split("\\r\\n");
+		Pattern pattern = Pattern.compile("^"+header, Pattern.CASE_INSENSITIVE);
+		for (String hdr : headerElements) {
+			Matcher matcher = pattern.matcher(hdr);
+			if (matcher.find()) {
+				String match = matcher.group();
+				matchedHeaders.add(match);
+			}
+		}
+		return matchedHeaders;
+	}
+	
+	private List<String> getAllowedWildcardSources(String policyText, Origin origin) {
+		
+		List<String> allowedSources = new ArrayList<String>();
+		Policy pol = ParserWithLocation.parse(policyText, origin);
+		
+		if (pol.allowsScriptFromSource(PARSED_WILDCARD_URI)) {
+			allowedSources.add("script-src");
+		}
+		if (pol.allowsStyleFromSource(PARSED_WILDCARD_URI)) {
+			allowedSources.add("style-src");
+		}
+		if (pol.allowsImgFromSource(PARSED_WILDCARD_URI)) {
+			allowedSources.add("img-src");
+		}
+		if (pol.allowsConnectTo(PARSED_WILDCARD_URI)) {
+			allowedSources.add("connect-src");
+		}
+		if (pol.allowsFrameFromSource(PARSED_WILDCARD_URI)) {
+			allowedSources.add("frame-src");
+		}
+		if (pol.allowsFrameAncestor(PARSED_WILDCARD_URI)) {
+			allowedSources.add("frame-ancestor");
+		}
+		if (pol.allowsFontFromSource(PARSED_WILDCARD_URI)) {
+			allowedSources.add("font-src");
+		}
+		if (pol.allowsMediaFromSource(PARSED_WILDCARD_URI)) {
+			allowedSources.add("media-src");
+		}
+		if (pol.allowsObjectFromSource(PARSED_WILDCARD_URI)) {
+			allowedSources.add("object-src");
+		}
+		if (pol.allowsManifestFromSource(PARSED_WILDCARD_URI)) {
+			allowedSources.add("manifest-src");
+		}
+		return allowedSources;
+	}
+	
+	@Override
+	public int getPluginId() {
+		return PLUGIN_ID;
+	}
+
+	@Override
+	public String getName() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "name");
+	}
+
+	private String getSolution() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "soln");
+	}
+
+	private String getReference() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "refs");
+	}
+
+	private void raiseAlert(HttpMessage msg, String name, int id, String description, String param, int risk) {
+		String alertName = StringUtils.isEmpty(name) ? getName() : getName() + ": " + name;
+		
+		Alert alert = new Alert(getPluginId(), risk, Alert.CONFIDENCE_MEDIUM, // PluginID, Risk, Reliability
+				alertName);
+		alert.setDetail(description, // Description
+				msg.getRequestHeader().getURI().toString(), // URI
+				param, // Param
+				"", // Attack
+				"", // Other info
+				getSolution(), // Solution
+				getReference(), // References
+				"", // Evidence

```

Comment:
Thoughts? Header field as:
- param (It was done this way because someone mentioned that it had been done elsewhere...)
- evidence (So it gets highlighted)
- both